### PR TITLE
add Dockerfile for ArchSpec

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
Signed-off-by: Carlos Eduardo Arango Gutierrez <carangog@redhat.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+USER root
+
+RUN dnf update -y && dnf install git python38 python3-pip -y
+RUN pip3 install poetry tox-gh-actions coverage
+RUN pip3 install archspec
+RUN dnf clean all
+
+RUN useradd archer
+USER archer
```
Comment (boegel) [inline Dockerfile:14]: Who's `archer`?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
+FROM registry.access.redhat.com/ubi8/python-38
+
+WORKDIR /archspec
+COPY . .
+
+USER root
+
+RUN dnf update -y && dnf install git python38 python3-pip -y
+RUN pip3 install poetry tox-gh-actions coverage
+RUN pip3 install archspec
```
Comment (boegel) [inline Dockerfile:10]: Maybe try running it too?

`archspec cpu`

**Review by boegel — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (ArangoGutierrez) [reply Dockerfile:14]: no one really, just a name that came to my head when assigning a non-root user to run archspec

File: Dockerfile (hunk 2 above)
Comment (ArangoGutierrez) [reply Dockerfile:10]: why? this is not a test container

Comment (boegel) [reply Dockerfile:10]: To prove it works. ;)

I guess the point of this container isn't entirely clear to me...

Comment (tgamblin) [reply Dockerfile:10]: my impression is that this is a base image for people to use if they want to build something that runs archspec.

Comment (ArangoGutierrez) [reply Dockerfile:10]: yup, people like me that don't run Python on a daily basis, I just want to get the best of Spack/ArchSpec without having to install pip on my machine

File: Dockerfile (hunk 1 above)
Comment (ArangoGutierrez) [reply Dockerfile:14]: I am open to change the running user to what ever name seems more appropriate.
`ArchSpec-runner` ?

Comment (tgamblin) [reply Dockerfile:14]: How about just archspec?

Comment (ArangoGutierrez) [reply Dockerfile]: afd85e7

**Review by tgamblin — DISMISSED** [review]

**Comment by tgamblin** [comment]
up to @boegel now

**Comment by ArangoGutierrez** [comment]
*PING* @boegel

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,16 @@
+FROM registry.access.redhat.com/ubi8/python-38
```
Comment (boegel) [inline Dockerfile:1]: What kind of base image is this? Why this one? Why not just CentOS?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+WORKDIR /archspec
+COPY . .
```
Comment (boegel) [inline Dockerfile:4]: What does this do?

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+USER root
+
+RUN dnf update -y && dnf install git python38 python3-pip -y
+RUN pip3 install poetry tox-gh-actions coverage
```
Comment (boegel) [inline Dockerfile:9]: I don't mind having these installed, but they're overkill, you don't need these to just run `archspec`

**Review by boegel — APPROVED** [review]

**Comment by alalazo** [comment]
Now that we have a Dockerfile, do we want an "Archspec" organization on Dockerhub where to publish container images? Or do we want to hook with [Github Container Registry](https://github.blog/2020-09-01-introducing-github-container-registry/)?

**Comment by ArangoGutierrez** [comment]
I will vote for the GitHub repo or Quay.io, but I may be biased so is up to Todd and Kenneth

**Pull request merged**